src/drivers/usb/usb_hub.c

```diff
@@ -62,27 +68,37 @@ struct usb_dev *usb_new_device(struct usb_dev *parent,
 	}
 
 	if (parent) { /* It's not a Root Hub */
+		int cfg;
+
 		hub = usb_dev_to_hub(parent);
 		assert(hub);
 		/* Here we reset device and set address. */
 		if (usb_hub_port_init(hub, dev, port) < 0) {
 			log_error("usb_hub_port_init failed");
 			goto out_err;
 		}
-		/* Fill device configuration. */
-		if (usb_get_configuration(dev, 0) < 0) {
-			log_error("usb_get_configuration failed");
-			goto out_err;
-		}
+
+		cfg = -1;
+		do {
+			if (++cfg > 0) {
+				usb_free_configuration(dev);
+			}
+			/* Fill device configuration. */
+			if (usb_get_configuration(dev, cfg) < 0) {
+				log_error("usb_get_configuration failed");
+				goto out_err;
+			}
+		/* Skip Microsoft's RNDIS */
+		} while (is_rndis(dev->iface_desc[0]));
```

Comment:
If device is RNDIS we call usb_free_configuration() after usb_get_configuration() ?
Or it means that RNDIS use cfg 1 instead 0?

Reply:
Yes, if device is RNDIS we free current configuration (rndis) and try to get next configuration. We repeat this until we find not rndis configuration and then set up it.